### PR TITLE
feat(skore/checks): Add baseline comparison checks (SKD009, SKD010)

### LINKED ISSUE #2668: Add model checks with baseline comparisons
Implement the following checks:

|Failure | Description | Typical signal |
|---|---|---|
|Model worse than baseline | The model performs worse than a sensible baseline | Performance metrics are on par or worse than baseline across the board|
|Model slower than baseline | The model is much slower than a sensible baseline for limited performance gain | Performance metrics are not significantly better than baseline across the board|

### PR DESCRIPTION
**Pull request by GaetandeCast**
#### Change description

Adds two new automated checks to `EstimatorReport.checks.summarize` (and via aggregation to `CrossValidationReport` / `ComparisonReport`):

- **SKD009 — Model worse than baseline**: flags when test scores are not significantly better than a `skrub.tabular_pipeline` wrapped `HistGradientBoostingClassifier` / `HistGradientBoostingRegressor` baseline (same train/test data, same metric set). Vote per non-timing metric, strict majority triggers the issue.
- **SKD010 — Model slower than baseline**: flags when `fit_time_` is at least 2x the fit time of a `skrub.tabular_pipeline` wrapped fast linear baseline (`LogisticRegression` for classification, `RidgeCV` for regression), AND the strict-majority "not significantly better than baseline" vote holds on test metrics. A 0.05s absolute floor on the time gap avoids noise on very fast fits.

Both checks raise `CheckNotApplicable` for unsupported ml tasks (multioutput, clustering, unknown) and when train+test data is unavailable. They share a single `_baseline_estimator_report(report, kind=...)` factory with kinds `"dummy"` / `"performance"` / `"fast"` — also used by the existing `SKD002` underfitting check.

Companion refactor of `SKD001` / `SKD002` / `SKD003` to read `MetricsSummaryDisplay.rows` directly (the `MetricsSummaryRow` `TypedDict`) instead of parsing the summary DataFrame. Metrics are now aligned by an identity tuple `(verbose_name, label, average, output)` rather than relying on row-order between two `summarize()` calls, which also lets `_TIMING_METRICS` shrink to the two canonical `Metric.verbose_name` values.

Closes #2668.

#### Contribution checklist

- [x] Unit tests were added or updated (if necessary)
- [x] Documentation was added or updated (if necessary)
- [x] The code passes our style conventions (you can check this by running pre-commit on your code with \`pre-commit run --all-files\`)
- [x] All the tests pass (please test locally before pushing)
- [x] The documentation builds and renders properly (if it does, our bot will add a comment linking to a preview of the documentation to review it visually)
- [x] A new changelog entry was added to CHANGELOG.rst (if necessary; typically, if your change requires updating tests)
- [x] All the commits in the PR are signed (more information [here](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits))
- [x] The pull request title respects the Conventional Commits convention (more information [here](https://docs.skore.probabl.ai/stable/contributing.html#pull-requests))

#### AI usage disclosure

AI tools were involved for:
- [x] Code generation (e.g., when writing an implementation or fixing a bug)
- [x] Test/benchmark generation
- [x] Documentation (including examples)
- [x] Research and understanding

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: sphinx/user_guide/automated_checks.rst [hunk 1]
```diff
@@ -347,3 +347,76 @@ How to reduce the risk
 - use regularization (Lasso, ElasticNet) to let the model select among
   correlated features,
 - group correlated features together before inspecting feature importance.
+
```
Comment (auguste-probabl) [inline sphinx/user_guide/automated_checks.rst:350]: Just a thought: this is the kind of documentation that can quickly get de-synced from the code.

File: skore/src/skore/_sklearn/_checks/model_checks.py [hunk 2]
```diff
@@ -183,7 +220,7 @@ def check_function(self, report: _BaseReport) -> str | None:
             [
                 detect_outliers_modified_zscore(report_data.loc[idx])
                 for idx in report_data.index
-                if idx not in _TIMING_METRICS
+                if idx not in ["fit_time_s", "predict_time_s"]
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/_checks/model_checks.py]: Could be factored to another TIMING_METRICS dict maybe

File: skore/src/skore/_sklearn/_checks/_utils.py [hunk 3]
```diff
@@ -1,22 +1,44 @@
...
 if TYPE_CHECKING:
     from skore._sklearn._estimator.report import EstimatorReport
+    from skore._sklearn._plot.metrics.metrics_summary_display import (
+        MetricsSummaryRow,
+    )
+    from skore._sklearn.types import DataSource
+
+_TIMING_METRICS = {"Fit time (s)", "Predict time (s)"}
+
+MetricKey = tuple[str, Any, Any, Any]
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/_checks/_utils.py]: nit
```suggestion
MetricName = str
Label = str
Average = str
Output = str
MetricKey = tuple[MetricName, Label, Average, Output]
```

**Review by auguste-probabl — DISMISSED** [review]
Some nitpicks but this is pretty much good to go!

File: sphinx/user_guide/automated_checks.rst (hunk 1 above)
Comment (GaetandeCast) [reply sphinx/user_guide/automated_checks.rst:350]: In what sense ? Like if we change how we define e.g. the baselines in the future ?

Comment (auguste-probabl) [reply sphinx/user_guide/automated_checks.rst:350]: Yes, it's pretty much an English-language implementation at this point. To be clear, this comment is not blocking: it's great to have detailed docs in principle, but it makes me scared that it eventually becomes wrong--and misleading docs is worse than no docs.

Comment (jeromedockes) [reply sphinx/user_guide/automated_checks.rst:350]: not sure I understand -- this list is recommendations on what users can do to reduce dimensionality, not a description of what skore does so I don't really see how it would become wrong

Comment (GaetandeCast) [reply sphinx/user_guide/automated_checks.rst:350]: He's referring  to the blocks below I think, they describe the choice of baselines.

**Review by jeromedockes — APPROVED** [review]
